go/src/koding/klient/machine/addr.go

```diff
@@ -0,0 +1,113 @@
+package machine
+
+import (
+	"encoding/json"
+	"errors"
+	"sync"
+	"time"
+)
+
+var (
+	// ErrAddrNotFound indicates that provided address does not exist.
+	ErrAddrNotFound = errors.New("address not found")
+)
+
+// IsAddrNotFound is a helper function that checks if provided error describes
+// missing address.
+func IsAddrNotFound(err error) bool {
+	return err == ErrAddrNotFound
+}
+
+// Addr satisfies net.Addr interface. It stores external machine address and
+// a time-stamp which indicates when the address was last seen being valid.
+type Addr struct {
+	// Net represents address network like "ip", "kite" etc.
+	Net string `json:"network"`
+
+	// Val stores a string representation of an address.
+	Val string `json:"value"`
+
+	// Updated stores the address update time.
+	Updated time.Time `json:"updated"`
```

Comment:
Usually we times with `At` suffix - `UpdateAt`, `CreatedAt` etc. Just for consistency's sake.